src/neo4j_genai/exceptions.py

```diff
@@ -0,0 +1,61 @@
+#  Copyright (c) "Neo4j"
+#  Neo4j Sweden AB [https://neo4j.com]
+#  #
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  #
+#      https://www.apache.org/licenses/LICENSE-2.0
+#  #
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+
+class RetrieverInitializationError(Exception):
```

Comment:
Should this class inherit from `ValidationError` instead?

Reply:
> 'ValidationError' is marked as '@final' and should not be subclassed 
Looks like it can't be subclassed. What is the concern with using `Exception` here?

Reply:
Ok. I was just wondering because we have to manually deal with the `errors`, subclassing would avoid that I guess, but not a big issue.